doc/helpers/VisualState-extensions.md

```diff
@@ -0,0 +1,110 @@
+# VisualStateExtensions Attached Properties
+Exposes visual states as attachable properties that lets you bind a string on a `Control` to set its `VisualState`s.
+
+## Remarks
+`VisualStateManager.GoToState` is typically used with `Control` where you would set `<VisualStateManager.VisualStateGroups>` on the root element of the ControlTemplate. Because this class is implemented using the same method, it means that if you are setting `Group1State` on an element, the `VisualStateManager.VisualStateGroups` should not be set on the very same element, but its first child:
+```xml
+<Page utu:VisualStateExtensions.Group1State="{Binding OnboardingState}">
+    <!-- Wrong -->
+    <VisualStateManager.VisualStateGroups>...
+
+    <!-- Good -->
+    <Grid>
+        <VisualStateManager.VisualStateGroups>...
+```
+
+Multiple `Group1..4State`s are provided:
+1. to allow for multiple concurrent/parallel states (eg: LoginStates and NetworkStates groups) where each group affect different sets of UI elements' proties; and
+2. to leave room for composite states where the activation is based on a specific combination (eg: LoggedIn + OnMeteredNetwork)
+
+## Properties
+Property|Type|Description
+-|-|-
+Group1State|string|Gets or sets the current state of the VisualStateGroup 1
+Group2State|string|Gets or sets the current state of the VisualStateGroup 2
+Group3State|string|Gets or sets the current state of the VisualStateGroup 3
+Group4State|string|Gets or sets the current state of the VisualStateGroup 4
```

Comment:
Just so I understand, why are we limiting to 4? Just because there aren't usually more than 4 VisualStateGroups in a template?

Reply:
I may have missed it in the discussions but would it be possible to just use a string collection so we can just do something like `.States="Focused, Pressed, etc."`

Reply:
~~Chosen by fair dice roll.~~ Joke aside, the original had 4 sets, which I think is enough for most cases. You would normally need 1-2.
It is feasible, and that one was part of [ViewDebugger's contract](#332). This syntax was just not chosen for `VisualStateExtensions`.

Reply:
On a second thought, we will be using the aggregate syntax.